spark/src/main/scala/geotrellis/spark/costdistance/IterativeCostDistance.scala

```diff
@@ -0,0 +1,221 @@
+ */
+
+package geotrellis.spark.costdistance
+
+import geotrellis.proj4.LatLng
+import geotrellis.raster._
+import geotrellis.raster.costdistance.CostDistance
+import geotrellis.spark._
+import geotrellis.vector._
+
+import org.apache.log4j.Logger
+import org.apache.spark.rdd.RDD
+import org.apache.spark.SparkContext
+import org.apache.spark.storage.StorageLevel
+import org.apache.spark.util.AccumulatorV2
+
+import scala.collection.mutable
+
+
+/**
+  * This Spark-enabled implementation of the standard cost-distance
+  * algorithm mentioned in the "previous work" section of [1] is
+  * "heavily inspired" by the MrGeo implementation [2] but does not
+  * share any code with it.
+  *
+  * 1. Tomlin, Dana.
+  *    "Propagating radial waves of travel cost in a grid."
+  *    International Journal of Geographical Information Science 24.9 (2010): 1391-1413.
+  *
+  * 2. https://github.com/ngageoint/mrgeo/blob/0c6ed4a7e66bb0923ec5c570b102862aee9e885e/mrgeo-mapalgebra/mrgeo-mapalgebra-costdistance/src/main/scala/org/mrgeo/mapalgebra/CostDistanceMapOp.scala
+  */
+object IterativeCostDistance {
+
+  type KeyCostPair = (SpatialKey, CostDistance.Cost)
+  type Changes = mutable.ArrayBuffer[KeyCostPair]
+
+  val logger = Logger.getLogger(IterativeCostDistance.getClass)
+
+  /**
+    * An accumulator to hold lists of edge changes.
+    */
+  class ChangesAccumulator extends AccumulatorV2[KeyCostPair, Changes] {
+    private val list: Changes = mutable.ArrayBuffer.empty
+
+    def copy: ChangesAccumulator = {
+      val other = new ChangesAccumulator
+      other.merge(this)
+      other
+    }
+    def add(pair: KeyCostPair): Unit = {
+      this.synchronized { list.append(pair) }
+    }
+    def isZero: Boolean = list.isEmpty
+    def merge(other: AccumulatorV2[KeyCostPair, Changes]): Unit =
+      this.synchronized { list ++= other.value }
+    def reset: Unit = this.synchronized { list.clear }
+    def value: Changes = list
+  }
+
+  def computeResolution[K: (? => SpatialKey), V: (? => Tile)](
+    friction: RDD[(K, V)] with Metadata[TileLayerMetadata[K]]
+  ) = {
+    val md = friction.metadata
+    val mt = md.mapTransform
+    val kv = friction.first
+    val key = implicitly[SpatialKey](kv._1)
+    val tile = implicitly[Tile](kv._2)
+    val extent = mt(key).reproject(md.crs, LatLng)
+    val degrees = extent.xmax - extent.xmin
+    val meters = degrees * (6378137 * 2.0 * math.Pi) / 360.0
+    val pixels = tile.cols
+    math.abs(meters / pixels)
+  }
+
+  /**
+    * Perform the cost-distance computation.
+    *
+    * @param  friction  The friction layer; pixels are in units of "seconds per meter"
+    * @param  points    The starting locations from-which to compute the cost of traveling
+    * @param  maxCost   The maximum cost before pruning a path (in units of "seconds")
+    */
+  def apply[K: (? => SpatialKey), V: (? => Tile)](
+    friction: RDD[(K, V)] with Metadata[TileLayerMetadata[K]],
+    points: Seq[Point],
```

Comment:
I think that there would be some mechanical changes needed in order to support an RDD of points, probably the friction layer and the points would need to be `join`ed (either logically or literally) as the first step.

As a practical matter, I do not think that supporting RDDs of points would add any real capability.  If someone really did have so many points that they do not fit into memory on one machine *and* some appreciable percentage of those points have unique projections into the raster, then it is likely that the raster layer is so large (in terms of pixels) that this algorithm would run out of driver memory before even reaching the second iteration.